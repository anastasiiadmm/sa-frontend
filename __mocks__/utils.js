import * as reduxHooks from 'react-redux';

jest.mock('../src/redux/store');
jest.mock('react-redux');

export const mockedUseSelectors = jest.spyOn(reduxHooks, 'useSelector');
export const mockedDispatch = jest.spyOn(reduxHooks, 'useDispatch');
