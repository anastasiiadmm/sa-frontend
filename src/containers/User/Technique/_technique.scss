@use 'assets/scss/_themes.scss';
@use 'assets/scss/_mixins.scss';

.Technique {
  &__table {
    padding: 16px;
    background: themes.$white-color;
  }

  &__name-column-style-text {
    position: absolute;
    left: 20px;
  }

  &__card-block {
    padding-bottom: 15px;
  }

  &__title {
    display: flex;
    justify-content: space-between;
    align-items: center;
    > h3 {
      margin: 0;
    }

    margin-bottom: 28px;
  }

  &__view-button {
    background: #ecf4ff;
    border: none;
    color: #1358bf;
    font-weight: 700;
    font-size: 14px;
  }

  &__card-style {
    height: 100%;
  }

  &__card-style-blocks {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
  }

  &__table_mobile {
    display: none;
  }

  &__card-style-cloud {
    background: themes.$cloud-color;
  }

  &__card-style-cloud-blocks {
    display: flex;
    flex-direction: row;
    gap: 12px;
  }

  &__card-style-cloud-button {
    margin-top: 5px;
    background: themes.$cloud-bg-button;
    color: themes.$info-color;
    font-weight: 700;
    font-size: 12px;
    border: transparent;
    padding: 0 10px;
  }

  &__image-styles {
    width: 48px;
    height: 50px;
    overflow: clip;
    border-radius: 3px;
    object-fit: cover;
  }

  &__card-title {
    font-size: 12px !important;
    font-weight: 700 !important;
    text-transform: uppercase;
    margin-bottom: 18px !important;
  }

  &__card-content {
    display: flex;
    gap: 30px;
    margin-bottom: 12px;

    > p {
      margin: 0;
      font-size: 24px;
      font-weight: 700;
    }
  }

  &__card-title {
    margin-right: 10px;
  }

  &__logo {
    display: none;
  }

  &__card-technique-block {
    display: none;
  }

  &__cloud_block_style {
    display: none;
  }

  &__mobile_data {
    display: none;
  }

  @include mixins.bp(mobile) {
    &__logo {
      display: none;
    }

    &__card-technique-block {
      display: none;
    }

    &__card-style-blocks {
      position: absolute;
      top: 70px;
      width: 100% !important;
      left: 0;
      right: 0;
      > div {
        width: 100%;
        margin-right: 20px;
        margin-left: 20px;
        height: 94px;
      }
    }

    &__card-style-cloud {
      width: 100% !important;
    }

    &__title {
      display: none;
    }
    &__mobile_data {
      display: block;
      .tractor {
        border-bottom: 1px solid #f5f7fa;
        height: 32px;
        display: flex;
        align-items: center;
        padding-bottom: 10px;
        margin-top: 0;
        .text_mobile_header {
          padding-left: 10px;
          color: #6c7994;
          font-size: 11px;
          > b {
            font-size: 14px;
          }
        }
      }
    }

    &__btn_mobile {
      > div {
        display: flex;
        justify-content: space-between;
        flex-wrap: nowrap!important;
        gap: 10px;

        > button {
          max-width: 100%;
        }
      }
    }

    &__card_btn_technique {
      display: none!important;
    }

    &__title_drawer {
      margin-left: -30px;
    }

    &__btn_mobile {
      margin-top: 84px;
      > h3 {
        font-weight: 500;
      }

      > div {
        display: flex;
        flex-wrap: wrap;
        gap: 10px;

        > button {
          border: none;
          max-width: 190px;
          width: 100%;
          height: 104px;
          text-align: left;
          background-color: #f8f8f8;
          border-radius: 12px;
          padding: 14px;
          > p {
            margin-top: 0px;
            font-size: 11px!important;
            font-weight: 500;
          }
        }
      }
    }

    &__table {
      display: none;
    }

    &__card-style-cloud-button {
      top: 10px;
      left: 34px;
    }

    &__cloudy_img {
      margin-top: -13px;
    }

    &__card-style-cloud-button_meteo {
      top: 32px;
      left: 32px;
    }

    .meteo-title_cloudy {
      position: absolute;
      top: 0;
      left: 57px;
    }

    .meteo-title {
      position: absolute;
      top: 36px;
      left: 57px;
    }

    &__card-style-cloud-button_meteo:disabled {
      position: absolute;
      top: 30px;
      left: 32px;
    }

    &__cloudy_img_not {
      top: 16px;
    }
    &__table_mobile {
      margin-left: 20px;
      margin-right: 20px;
      margin-bottom: 70px;
      display: block;

      > div {
        width: 100%;
        background-color: #ffffff;
        padding: 16px;
        margin-bottom: 10px;
        border-radius: 12px;

        .table_img {
          display: flex;
          > div {
            > img {
              width: 32px;
              height: 37px;
              border-radius: 8px;
            }
          }

          .text_table {
            line-height: 0.5;
            margin-top: -10px;
            padding-left: 15px;
            > p {
              > b {
                font-weight: 500;
                color: #161a24;
              }

              color: #79859e;
            }
          }
        }
      }
    }
    &__tractor_icon_mobile {
      border-radius: 50%;
    }

    &__card-style-cloud-blocks {
      position: relative;
    }

    &__card-style-cloud {
      height: 44px!important;
    }
    &__card-style-cloud-button:disabled {
      position: absolute;
      top: -24px;
      left: 15px;
    }

    &__cloudy_img_not {
      margin-top: -20px;
    }

    .meteo-title_cloudy {
      margin-top: -20px!important;
    }
    .meteo-title {
      margin-top: -40px!important;
    }

    &__card-style-cloud-button_meteo {
      background: linear-gradient(103.37deg, #3d9fff 2.9%, #1785f0 64.65%) !important;
    }
    &__card-style-cloud-button_meteo {
      width: 160px;
      left: -23px !important;
      top: 22px !important;
      height: 24px!important;
      font-size: 10px!important;
    }
    &__card-style-cloud-button_meteo:disabled {
      background-color: #7E848F!important;
    }
  }

  @include mixins.bp(desktop) {
    &__card-style-blocks {
      justify-content: left;
    }
    &__logo {
      display: block;
      position: absolute;
      top: 20px;
      left: 20px;
      z-index: 200;

      > img {
        width: 96px;
        height: 40px;
      }
    }

    &__card-technique {
      display: none;
    }

    &__card-technique-block {
      display: block;
      width: 214px;
      background-color: #f5f6ff;
      padding: 14px 16px 0px 12px;
      border-radius: 12px;
      margin-right: 20px;
      .logo_mobile {
        width: 24px;
        height: 24px;
        background-color: white;
        display: flex;
        align-items: center;
        justify-content: center;
        border-radius: 30px;

        > img {
          width: 16px;
          height: 16px;
        }
      }

      > p {
        margin-top: 3px;
      }
    }

    &__card-content {
      margin-top: -15px;
      > p {
        font-size: 13px;
      }
    }

    &__cloud_block {
      display: none;
    }

    &__cloud_block_style {
      display: block;
    }

    &__card-style-cloud {
      background: linear-gradient(103.37deg, #3d9fff 2.9%, #1785f0 64.65%);
    }
    &__cloudy_img_not {
      position: absolute;
      top: 8px;
      left: 15px;
    }

    .meteo-title_cloudy {
      color: #ffffff;
      font-weight: 400;
      font-size: 12px;
      margin-left: -10px;
      margin-top: 6px;
    }
    .meteo-title {
      color: #ffffff;
      font-size: 12px;
      margin-left: -10px;
      margin-top: -16px;
      font-weight: 600;
    }

    &__cloudy_img {
      margin-left: -10px;
    }

    &__card-style-cloud-button_meteo {
      position: absolute;
      left: 0;
      bottom: 2px;
      background: transparent;
      border: none;
      color: #ffffff;
      font-weight: 500;
      font-size: 12px;
    }

    &__card-style-cloud-button_meteo:disabled {
      position: absolute;
      left: 0;
      bottom: 2px;
      background: transparent;
      border: none;
      color: #ffffff;
      font-weight: 500;
      font-size: 12px;
    }

    &__card-style-cloud-button {
      position: absolute;
      left: 0;
      top: 40px;
      background: transparent !important;
      font-size: 11px;
    }

    &__card-style-cloud-button:disabled {
      background: transparent !important;
      color: #ffffff;
    }

    &__title {
      display: block;
      padding-left: 10px;
      > h3 {
        margin-left: -10px;
      }
    }

    &__btn_title {
      margin-top: 10px;
      > button {
        width: 242px;
        height: 40px;
        border-radius: 8px;
        border: none;
        box-shadow: none;
      }
    }

    &__btn_click {
      display: flex;
      margin-top: 5px;
      > button {
        background-color: transparent;
        border: none;
        padding: 0;
        margin-right: 10px;
        > img {
          width: 36px;
          height: 36px;
        }
      }
    }

    &__view-button {
      width: 242px;
      height: 40px;
      background-color: #f8f8f8;
      border-radius: 8px;
      color: #689f3a;
    }

    &__title_drawer {
      font-weight: 500;
      font-size: 15px;
      color: #1d2024;
      text-align: center;
      margin: 0;
      padding: 0;
    }
  }
}

.meteo-h1 {
  margin: 0 !important;
  padding: 0;
}

.send-button {
  background: #0078a8;
}
