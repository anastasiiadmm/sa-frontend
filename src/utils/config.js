export const apiURL = '';

export const defaultError = { detail: 'На сервере что-то пошло не так' };
