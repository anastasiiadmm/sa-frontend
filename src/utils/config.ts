import { AxiosRequestConfig, AxiosRequestHeaders } from 'axios';

import { checkForTokens, logoutUser } from 'redux/auth/authSlice';
import store from 'redux/store';
import { TSetupAxiosInterceptor } from 'type';

import { apiUrlsV2, serverUrls, serverUrlsSocket } from 'utils/constants';
import { logoutLocalStorage } from 'utils/token';

const { REACT_APP_ENVIRONMENT } = process.env;

const appEnvironment = REACT_APP_ENVIRONMENT || 'local';

export const apiURL2 = apiUrlsV2[appEnvironment];
export const socketApiSocket = serverUrlsSocket[appEnvironment];

export const apiUrlCrop = serverUrls[appEnvironment];

export const setupAxiosInterceptors: TSetupAxiosInterceptor = (axiosInstance) => {
  return axiosInstance.interceptors.request.use(async (config: AxiosRequestConfig) => {
    const key = store.getState()?.auth?.tokens?.access;

    if (key) {
      config.headers = {
        ...config.headers,
        Authorization: `Bearer ${key}`,
      } as AxiosRequestHeaders;
    }
    return config;
  });
};

export const setupAxiosInterceptorsResponse: TSetupAxiosInterceptor = (axiosInstance) => {
  return axiosInstance.interceptors.response.use(
    async (config) => {
      return config;
    },
    async (error) => {
      const originalRequest = error.config;

      const statusCode = error?.response?.status;
      const index = store;
      const { tokens } = index.getState().auth;
      if (
        tokens?.access &&
        statusCode === 401 &&
        error.config &&
        !error.config._isReady &&
        error.response.data.messages
      ) {
        originalRequest._isReady = true;
        try {
          const resp = await axiosInstance.post('/accounts/refresh/', { refresh: tokens.refresh });
          if (resp.status === 200) {
            const newTokens = resp.data;
            axiosInstance.defaults.headers.Authorization = `Bearer ${newTokens.access}`;
            const usersLocal = {
              access: resp.data.access,
              refresh: tokens.refresh,
              is_manager: tokens.is_manager,
            };
            index.dispatch(checkForTokens(usersLocal));
            const obj = {
              access: resp.data.access,
              is_manager: tokens.is_manager,
            };
            localStorage.setItem('users', JSON.stringify(obj));
            window.dispatchEvent(new Event('storage'));
            return axiosInstance(originalRequest);
          }
        } catch (e) {
          logoutLocalStorage();
          window.location.reload();
        }

        store.dispatch(logoutUser());
      }

      return Promise.reject(error);
    },
  );
};
