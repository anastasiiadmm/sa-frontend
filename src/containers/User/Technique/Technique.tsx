import { EyeOutlined } from '@ant-design/icons';
import { Button, Card, Tooltip, Typography } from 'antd';
import { ColumnsType } from 'antd/es/table';
import bem from 'easy-bem';
import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';

import cloudy from 'assets/images/icons/cloudy.svg';
import tractorBlue from 'assets/images/icons/tractor-blue.svg';
import tractor from 'assets/images/icons/tractor-image.svg';
import AddUpdateTechnique from 'components/ModalComponent/ModalChildrenComponents/AddUpdateTechnique/AddUpdateTechnique';
import ModalComponent from 'components/ModalComponent/ModalComponent';
import SkeletonBlock from 'components/SkeletonBlock/SkeletonBlock';
import TableComponent from 'components/TableComponent/TableComponent';
import { accountsSelector, fetchUser, fetchUserVehicles } from 'redux/accounts/accountsSlice';
import { useAppDispatch, useAppSelector } from 'redux/hooks';
import { userVehicles } from 'types/types';
import { apiUrlCrop } from 'utils/config';
import 'containers/User/Technique/_technique.scss';

const { Title } = Typography;

const Technique = () => {
  const b = bem('Technique');
  const {
    user: userAccount,
    fetchLoadingUser,
    userVehicles,
    fetchUserVehiclesLoading,
    userVehiclesPagination,
  } = useAppSelector(accountsSelector);
  const dispatch = useAppDispatch();
  const [isModalOpen, setIsModalOpen] = useState(false);
  const [filters, setFilters] = useState({
    page: 1,
  });

  useEffect(() => {
    dispatch(fetchUser());
  }, [dispatch]);

  useEffect(() => {
    const data = {
      query: {
        page: filters?.page,
      },
    };

    dispatch(fetchUserVehicles({ data }));
  }, [dispatch, filters]);

  const showModal = () => {
    setIsModalOpen(true);
  };

  const handleOkCancel = () => {
    setIsModalOpen(!isModalOpen);
  };

  const pagePrevHandler = () => {
    setFilters({ ...filters, page: filters.page - 1 });
  };

  const pageNextHandler = () => {
    setFilters({ ...filters, page: filters.page + 1 });
  };

  const columns: ColumnsType<userVehicles> = [
    {
      key: 'code',
      title: 'Код техники',
      dataIndex: 'code',
      width: '20%',
      fixed: 'left',
      render: (text: string, record) => {
        return (
          <div style={{ display: 'flex', gap: 12 }}>
            <img
              className={b('image-styles')}
              src={record.image ? `${apiUrlCrop}${record.image}` : tractor}
              alt='tractor'
            />
            <p className={b('name-column-style')}>{record.code}</p>
          </div>
        );
      },
    },
    {
      key: 'description',
      title: 'Название',
      dataIndex: 'description',
      width: '30%',
    },
    {
      key: 'field_count',
      title: 'Поля',
      dataIndex: 'field_count',
      width: '20%',
      render: (text: number, record) => (
        <p className={b('name-column-style')}>{record?.vehicle_fields_data?.field_count}</p>
      ),
    },
    {
      key: 'processed_area',
      title: 'Общая Площадь',
      dataIndex: 'processed_area',
      width: '20%',
      render: (text: number, record) => (
        <p className={b('name-column-style')}>{record?.vehicle_fields_data?.processed_area}</p>
      ),
    },
    {
      dataIndex: 'profile',
      width: '30%',
      render: (text: string, record) => (
        <div style={{ display: 'flex', gap: 20, alignItems: 'center', justifyContent: 'right' }}>
          <Tooltip
            title='Просмотреть профиль'
            color='#BBBBBB'
            overlayInnerStyle={{ padding: '5px 15px', borderRadius: 15 }}
          >
            <Link to={`/profile-technique/${userAccount?.id}/${record.id}`}>
              <Button type='text'>
                <img style={{ width: 27 }} src={tractorBlue} alt='tractorBlue' />
              </Button>
            </Link>
          </Tooltip>
          <Tooltip
            title='Просмотреть на карте'
            color='#BBBBBB'
            overlayInnerStyle={{ padding: '5px 15px', borderRadius: 15 }}
            placement='topRight'
          >
            <Link to='/'>
              <Button type='text' style={{ display: 'flex', alignItems: 'center' }}>
                <EyeOutlined style={{ fontSize: '27px', color: '#1358bf' }} />
              </Button>
            </Link>
          </Tooltip>
        </div>
      ),
    },
  ];

  return (
    <>
      <div className={b()} data-testid='technique-id'>
        <div className={b('card-block')}>
          <Card className={b('card-style')} bordered={false}>
            <div className={b('card-style-blocks')}>
              <div>
                <Title className={b('card-title')}>Количество техники</Title>
                {fetchLoadingUser ? (
                  <SkeletonBlock active={fetchLoadingUser} num={1} titleBool />
                ) : (
                  <div className={b('card-content')}>
                    <img src={tractorBlue} alt='group' />
                    <p>{userAccount?.vehicles_amount}</p>
                  </div>
                )}
              </div>
              <div>
                <Card className={b('card-style-cloud')} bordered={false}>
                  <div className={b('card-style-cloud-blocks')}>
                    <div>
                      <img src={cloudy} alt='cloudy' />
                    </div>
                    <div>
                      <Title className={b('card-title meteo-h1')}>Подключите метеосервис</Title>
                      <p className='meteo-h1'>Все про погоду и почву</p>
                      <Button className={b('card-style-cloud-button')}>Отправить запрос</Button>
                    </div>
                  </div>
                </Card>
              </div>
            </div>
          </Card>
        </div>
        <div className={b('table')}>
          <div className={b('title')}>
            <Title level={3} data-testid='sign_in_test' className={b('title')}>
              Техника
            </Title>

            <Button type='primary' onClick={showModal}>
              Запрос на добавление техники
            </Button>
          </div>

          <TableComponent
            loading={fetchUserVehiclesLoading}
            columns={columns}
            data={userVehicles}
            rowKey={(record) => record.id as number}
            params={userVehiclesPagination}
            pagePrevHandler={pagePrevHandler}
            pageNextHandler={pageNextHandler}
          />
        </div>
      </div>

      <ModalComponent
        title='Добавить технику'
        open={isModalOpen}
        handleOk={handleOkCancel}
        handleCancel={handleOkCancel}
      >
        <AddUpdateTechnique userId={null} isRequest handleEditOkCancel={handleOkCancel} />
      </ModalComponent>
    </>
  );
};

export default Technique;
