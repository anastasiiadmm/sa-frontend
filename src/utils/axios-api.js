import axios from "axios";
import {apiURL} from "utils/config";

export const axiosApi = axios.create({
  baseURL: apiURL,
});