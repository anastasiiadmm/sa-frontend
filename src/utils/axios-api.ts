import axios from 'axios';

import { checkForTokens, logoutUser } from 'redux/auth/authSlice';
import { apiURL } from 'utils/config';
import { logoutLocalStorage } from 'utils/token';
import store from 'redux/store';

const axiosApi = axios.create({
  baseURL: apiURL,
});

axiosApi.interceptors.request.use(async (config) => {
  const key = store.getState()?.auth?.tokens?.access;

  if (key) {
    config.headers = {
      ...config.headers,
      Authorization: `Bearer ${key}`,
      'Content-Type': 'application/json',
    } as any;
  }
  return config;
});

axiosApi.interceptors.response.use(
  async (config) => {
    return config;
  },
  async (error) => {
    const originalRequest = error.config;

    const statusCode = error?.response?.status;
    const index = store;
    const { tokens } = index.getState().auth;
    if (
      tokens &&
      statusCode === 401 &&
      error.config &&
      !error.config._isReady &&
      error.response.data.messages
    ) {
      originalRequest._isReady = true;
      try {
        const resp = await axiosApi.post('/accounts/refresh/', { refresh: tokens.refresh });
        if (resp.status === 200) {
          const newTokens = resp.data;
          axiosApi.defaults.headers.Authorization = `Bearer ${newTokens.access}`;
          const usersLocal = {
            user: index.getState()?.auth?.user,
            token: {
              access: resp.data.access,
              refresh: tokens.refresh,
            },
          };
          index.dispatch(checkForTokens(usersLocal));
          const obj: any = usersLocal;
          delete obj.token.refresh;
          localStorage.setItem('users', JSON.stringify(obj));
          window.dispatchEvent(new Event('storage'));
          return axiosApi(originalRequest);
        }
      } catch (e) {
        logoutLocalStorage();
        window.location.reload();
      }

      store.dispatch(logoutUser());
    }

    return Promise.reject(error);
  },
);

export default axiosApi;
