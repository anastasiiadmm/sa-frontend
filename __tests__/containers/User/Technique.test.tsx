import { BrowserRouter } from "react-router-dom";
import {cleanup} from "@testing-library/react";
import "../../../__mocks__/matchMedia.mock";
import "@testing-library/jest-dom";
import { mockedDispatch, mockedUseSelectors } from "../../../__mocks__/utils";

import Technique from "../../../src/containers/User/Technique/Technique";
import * as TestRenderer from 'react-test-renderer';

afterEach(cleanup);

const data = {
    userVehicles: [
        {
            "id": 94,
            "code": "P4L7SG518",
            "description": "adasdasd213123",
            "jobs_number": 0,
            "area": "0",
            "image": "/media/files/inquiries/219/.jpeg"
        },
    ]
};


describe("<Technique />", () => {
    test("Users table component should be in the document", async () => {
        mockedUseSelectors.mockReturnValue(data);
        const dispatch = jest.fn();
        mockedDispatch.mockReturnValue(dispatch);

        const tree = TestRenderer
            .create(
                <BrowserRouter>
                    <Technique />
                </BrowserRouter>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
