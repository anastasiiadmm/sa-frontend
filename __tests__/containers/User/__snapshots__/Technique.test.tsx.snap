// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Technique /> Users table component should be in the document 1`] = `
<div
  className="Technique"
  data-testid="technique-id"
>
  <div
    className="Technique__card-block"
  >
    <div
      className="ant-card Technique__card-style css-dev-only-do-not-override-12upa3x"
    >
      <div
        className="ant-card-body"
        style={{}}
      >
        <div
          className="Technique__card-style-blocks"
        >
          <div>
            <h1
              className="ant-typography Technique__card-title css-dev-only-do-not-override-12upa3x"
              style={
                {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              Количество техники
            </h1>
            <div
              className="Technique__card-content"
            >
              <img
                alt="group"
                src={
                  {
                    "process": [Function],
                  }
                }
              />
              <p />
            </div>
          </div>
          <div>
            <div
              className="ant-card Technique__card-style-cloud css-dev-only-do-not-override-12upa3x"
              style={
                {
                  "width": 300,
                }
              }
            >
              <div
                className="ant-card-body"
                style={{}}
              >
                <div
                  className="Technique__card-style-cloud-blocks"
                >
                  <div>
                    <img
                      alt="cloudy"
                      src={
                        {
                          "process": [Function],
                        }
                      }
                    />
                  </div>
                  <div>
                    <h1
                      className="ant-typography Technique__card-title meteo-h1 css-dev-only-do-not-override-12upa3x"
                      style={
                        {
                          "WebkitLineClamp": undefined,
                        }
                      }
                    >
                      Подключите метеосервис
                    </h1>
                    <p
                      className="meteo-h1"
                    >
                      Все про погоду и почву
                    </p>
                    <button
                      className="ant-btn css-dev-only-do-not-override-12upa3x ant-btn-default Technique__card-style-cloud-button"
                      disabled={false}
                      onClick={[Function]}
                      style={{}}
                      type="button"
                    >
                      <span>
                        Отправить запрос
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="Technique__table"
  >
    <div
      className="Technique__title"
    >
      <h3
        className="ant-typography Technique__title css-dev-only-do-not-override-12upa3x"
        data-testid="sign_in_test"
        style={
          {
            "WebkitLineClamp": undefined,
          }
        }
      >
        Техника
      </h3>
      <div>
        <button
          className="ant-btn css-dev-only-do-not-override-12upa3x ant-btn-primary"
          disabled={false}
          onClick={[Function]}
          style={
            {
              "marginRight": 15,
            }
          }
          type="button"
        >
          <span>
            Запрос на добавление техники
          </span>
        </button>
        <a
          href="/technique-map"
          onClick={[Function]}
        >
          <button
            className="ant-btn css-dev-only-do-not-override-12upa3x ant-btn-default Technique__view-button"
            disabled={false}
            onClick={[Function]}
            style={{}}
            type="button"
          >
            <span>
              Посмотреть всю технику
            </span>
          </button>
        </a>
      </div>
    </div>
    <div
      className="ant-table-wrapper css-dev-only-do-not-override-12upa3x"
    >
      <div
        className="ant-spin-nested-loading css-dev-only-do-not-override-12upa3x"
      >
        <div
          className="ant-spin-container"
        >
          <div
            className="ant-table ant-table-scroll-horizontal"
          >
            <div
              className="ant-table-container"
            >
              <div
                className="ant-table-content"
                onScroll={[Function]}
                style={
                  {
                    "overflowX": "auto",
                    "overflowY": "hidden",
                  }
                }
              >
                <table
                  style={
                    {
                      "minWidth": "100%",
                      "tableLayout": "auto",
                      "width": 950,
                    }
                  }
                >
                  <colgroup>
                    <col
                      style={
                        {
                          "width": "20%",
                        }
                      }
                    />
                    <col
                      style={
                        {
                          "width": "30%",
                        }
                      }
                    />
                    <col
                      style={
                        {
                          "width": "20%",
                        }
                      }
                    />
                    <col
                      style={
                        {
                          "width": "28%",
                        }
                      }
                    />
                  </colgroup>
                  <thead
                    className="ant-table-thead"
                  >
                    <tr>
                      <th
                        className="ant-table-cell"
                        colSpan={null}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        rowSpan={null}
                        scope="col"
                        style={{}}
                      >
                        Код техники
                      </th>
                      <th
                        className="ant-table-cell"
                        colSpan={null}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        rowSpan={null}
                        scope="col"
                        style={{}}
                      >
                        Название
                      </th>
                      <th
                        className="ant-table-cell"
                        colSpan={null}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        rowSpan={null}
                        scope="col"
                        style={{}}
                      >
                        Задача на обработку
                      </th>
                      <th
                        className="ant-table-cell"
                        colSpan={null}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        rowSpan={null}
                        scope="col"
                        style={{}}
                      >
                        Общая Площадь
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    className="ant-table-tbody"
                  >
                    <tr
                      aria-hidden="true"
                      className="ant-table-measure-row"
                      style={
                        {
                          "fontSize": 0,
                          "height": 0,
                        }
                      }
                    >
                      <td
                        style={
                          {
                            "border": 0,
                            "height": 0,
                            "padding": 0,
                          }
                        }
                      >
                        <div
                          style={
                            {
                              "height": 0,
                              "overflow": "hidden",
                            }
                          }
                        >
                           
                        </div>
                      </td>
                      <td
                        style={
                          {
                            "border": 0,
                            "height": 0,
                            "padding": 0,
                          }
                        }
                      >
                        <div
                          style={
                            {
                              "height": 0,
                              "overflow": "hidden",
                            }
                          }
                        >
                           
                        </div>
                      </td>
                      <td
                        style={
                          {
                            "border": 0,
                            "height": 0,
                            "padding": 0,
                          }
                        }
                      >
                        <div
                          style={
                            {
                              "height": 0,
                              "overflow": "hidden",
                            }
                          }
                        >
                           
                        </div>
                      </td>
                      <td
                        style={
                          {
                            "border": 0,
                            "height": 0,
                            "padding": 0,
                          }
                        }
                      >
                        <div
                          style={
                            {
                              "height": 0,
                              "overflow": "hidden",
                            }
                          }
                        >
                           
                        </div>
                      </td>
                    </tr>
                    <tr
                      className="ant-table-row ant-table-row-level-0"
                      data-row-key={94}
                      onClick={[Function]}
                      style={{}}
                    >
                      <td
                        className="ant-table-cell"
                        colSpan={null}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        rowSpan={null}
                        style={{}}
                      >
                        <div
                          style={
                            {
                              "display": "flex",
                              "gap": 12,
                            }
                          }
                        >
                          <img
                            alt="tractor"
                            className="Technique__image-styles"
                            src="https://localhost:8000/media/files/inquiries/219/.jpeg"
                          />
                          <p
                            className="Technique__name-column-style"
                          >
                            P4L7SG518
                          </p>
                        </div>
                      </td>
                      <td
                        className="ant-table-cell"
                        colSpan={null}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        rowSpan={null}
                        style={{}}
                      >
                        adasdasd213123
                      </td>
                      <td
                        className="ant-table-cell"
                        colSpan={null}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        rowSpan={null}
                        style={{}}
                      >
                        <p
                          className="Technique__name-column-style"
                        >
                          0
                        </p>
                      </td>
                      <td
                        className="ant-table-cell"
                        colSpan={null}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        rowSpan={null}
                        style={{}}
                      >
                        <div
                          style={
                            {
                              "alignItems": "center",
                              "display": "flex",
                              "justifyContent": "right",
                            }
                          }
                        >
                          <p
                            className="Technique__name-column-style-text"
                          >
                            0
                          </p>
                          <a
                            href="/profile-technique/undefined/94"
                            onClick={[Function]}
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                          >
                            <button
                              className="ant-btn css-dev-only-do-not-override-12upa3x ant-btn-text"
                              disabled={false}
                              onClick={[Function]}
                              style={{}}
                              type="button"
                            >
                              <span
                                aria-label="eye"
                                className="anticon anticon-eye"
                                role="img"
                                style={
                                  {
                                    "color": "#1358bf",
                                    "fontSize": "27px",
                                  }
                                }
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="eye"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                                  />
                                </svg>
                              </span>
                            </button>
                          </a>
                          <a
                            href="/open-map/94/local-tractor/"
                            onClick={[Function]}
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                          >
                            <button
                              className="ant-btn css-dev-only-do-not-override-12upa3x ant-btn-text"
                              disabled={false}
                              onClick={[Function]}
                              style={
                                {
                                  "alignItems": "center",
                                  "display": "flex",
                                }
                              }
                              type="button"
                            >
                              <img
                                alt="tractorBlue"
                                src={
                                  {
                                    "process": [Function],
                                  }
                                }
                                style={
                                  {
                                    "width": 27,
                                  }
                                }
                              />
                            </button>
                          </a>
                        </div>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="pagination"
    >
      <div
        className="pagination__pagination_show"
      >
        <div
          className="glav_pagination"
        >
          <div
            className="next_table"
          >
            <button
              className="ant-btn css-dev-only-do-not-override-12upa3x ant-btn-default ant-btn-icon-only pagination_buttons pagination_buttons-prev"
              disabled={true}
              onClick={[Function]}
              style={{}}
              type="button"
            >
              <span
                className="ant-btn-icon"
                style={{}}
              >
                <img
                  alt="backIcons"
                  src={
                    {
                      "process": [Function],
                    }
                  }
                />
              </span>
            </button>
            <button
              className="ant-btn css-dev-only-do-not-override-12upa3x ant-btn-default ant-btn-icon-only pagination_buttons pagination_buttons-next"
              disabled={true}
              onClick={[Function]}
              role="button"
              style={{}}
              type="button"
            >
              <span
                className="ant-btn-icon"
                style={{}}
              >
                <img
                  alt="nextIcons"
                  src={
                    {
                      "process": [Function],
                    }
                  }
                />
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
